pkg/operator/ceph/csi/operator_config.go

```diff
@@ -0,0 +1,178 @@
+	}
+	logger.Info("Successfully updated ceph-CSI operator config CR")
+
+	return nil
+}
+
+func (r *ReconcileCSI) generateCSIOpConfigSpec(cluster cephv1.CephCluster, opConfig *csiopv1a1.OperatorConfig, imageSetCmName string) csiopv1a1.OperatorConfigSpec {
+	cephfsClientType := csiopv1a1.KernelCephFsClient
+	if CSIParam.ForceCephFSKernelClient == "false" {
+		cephfsClientType = csiopv1a1.AutoDetectCephFsClient
+	}
+
+	opConfig.Spec = csiopv1a1.OperatorConfigSpec{
+		Log: &csiopv1a1.OperatorLogSpec{
+			Verbosity: int(CSIParam.LogLevel),
+		},
+		DriverSpecDefaults: &csiopv1a1.DriverSpec{
+			Log: &csiopv1a1.LogSpec{
+				Verbosity: int(CSIParam.LogLevel),
+			},
+			ImageSet: &v1.LocalObjectReference{
+				Name: imageSetCmName,
+			},
+			ClusterName:      &cluster.Name,
+			EnableMetadata:   &CSIParam.CSIEnableMetadata,
+			GRpcTimeout:      int(CSIParam.GRPCTimeout),
+			GenerateOMapInfo: &CSIParam.EnableOMAPGenerator,
+			FsGroupPolicy:    k8scsiv1.FileFSGroupPolicy,
+			NodePlugin: &csiopv1a1.NodePluginSpec{
+				PodCommonSpec: csiopv1a1.PodCommonSpec{
+					PrioritylClassName: &CSIParam.ProvisionerPriorityClassName,
+					Affinity: &v1.Affinity{
+						NodeAffinity: getNodeAffinity(r.opConfig.Parameters, pluginNodeAffinityEnv, &v1.NodeAffinity{}),
+					},
+					Tolerations: getToleration(r.opConfig.Parameters, pluginTolerationsEnv, []v1.Toleration{}),
+				},
+				Resources:              csiopv1a1.NodePluginResourcesSpec{},
+				KubeletDirPath:         CSIParam.KubeletDirPath,
+				EnableSeLinuxHostMount: &CSIParam.EnablePluginSelinuxHostMount,
+			},
+			ControllerPlugin: &csiopv1a1.ControllerPluginSpec{
+				PodCommonSpec: csiopv1a1.PodCommonSpec{
+					PrioritylClassName: &CSIParam.PluginPriorityClassName,
+					Affinity: &v1.Affinity{
+						NodeAffinity: getNodeAffinity(r.opConfig.Parameters, provisionerNodeAffinityEnv, &v1.NodeAffinity{}),
+					},
+					Tolerations: getToleration(r.opConfig.Parameters, provisionerTolerationsEnv, []v1.Toleration{}),
+				},
+				Replicas:  &CSIParam.ProvisionerReplicas,
+				Resources: csiopv1a1.ControllerPluginResourcesSpec{},
+			},
+
+			LeaderElection: &csiopv1a1.LeaderElectionSpec{
+				LeaseDuration: int(CSIParam.LeaderElectionLeaseDuration),
+				RenewDeadline: int(CSIParam.LeaderElectionRenewDeadline),
+				RetryPeriod:   int(CSIParam.LeaderElectionRetryPeriod),
+			},
+			DeployCsiAddons:  &CSIParam.EnableCSIAddonsSideCar,
+			CephFsClientType: cephfsClientType,
+		},
+	}
+	if !reflect.DeepEqual(cluster.Spec.Network, cephv1.NetworkSpec{}) {
+
+		if cluster.Spec.Network.Connections.Encryption.Enabled {
+			opConfig.Spec.DriverSpecDefaults.Encryption.ConfigMapRef = v1.LocalObjectReference{
+				Name: "rook-ceph-csi-kms-config",
+			}
+		}
+	}
+
+	return opConfig.Spec
+}
+
+func (r *ReconcileCSI) createImageSetConfigmap() (string, error) {
+	cm := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-csi-operator-image-set-configmap",
```

Comment:
@subhamkrai Downstream uninstall steps might need to be updated to garbage collect the resources we are not deleting